components/Button.tsx

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import styled from 'styled-components/native';
+
+export interface ButtonProps {
+  /**
+   * Button's text
+   */
+  text: string;
+  /**
+   *  Option to fit button width to its parent width
+   */
+  block?: boolean;
+  /**
+   * Set the button type to outline
+   */
+  outline?: boolean;
+  /**
+   * Set the button to disabled state
+   */
+  disabled?: boolean;
+  /**
+   * Called when button is pressed
+   */
+  onPress: () => void;
+}
+
+export const Button = (props: ButtonProps) => {
+  const {text, block, outline, onPress, disabled} = props;
+  return (
+    <ButtonWithStyled
+      onPress={onPress}
+      outline={outline}
+      block={block}
+      disabled={disabled}>
+      <ButtonText outline={outline}>{text}</ButtonText>
+    </ButtonWithStyled>
+  );
+};
+
+const ButtonWithStyled = styled.TouchableOpacity<{
```

Comment:
Can we separate style and code file later on?

Reply:
Yes. After the redactor phase structure will be like

       root
       |- components
       |      |- index.ts
       |      |- Button
       |           |- Button.tsx
       |           |- Button.style.ts
       |- screens
              |- index.ts
              |- Home
                   |- Home.tsx
                   |- Home.style.ts

Reply:
Gotcha